### DISCUSSION TITLE
Flatcar on Xen

### DISCUSSION THREAD

**Discussion by benv666** [flatcar/Flatcar#657] [Q&A]
Hi folks,

Did anyone manage to actually boot Flatcar on Xen?
I'm running a self-compiled Xen 4.16 and was hoping to get Flatcar up and running, so I got myself the latest stable xen image (https://stable.release.flatcar-linux.net/amd64-usr/current/flatcar_production_xen_image.bin.bz2), which I bunzip2-ed and dd-ed to a disk partition.
Next I made a tiny config with pygrub and hoped to see it boot:
```
bootloader = "pygrub"
builder = "generic"
memory = 6144
name = "flatcar"
vif = [ ]
vcpus = 2
disk = ['drbd:flatcar-root,xvda1,w']
```

However, when trying to boot this I get the grub Menu, but when it tries to boot any entry I get a useless "error: not xen image"
When looking around in the grub console things look sane as far as I can judge them, but I don't really know what to expect.

Does anyone know how to proceed?

Thanks
![flatcar-1](https://user-images.githubusercontent.com/165034/156378600-435d8a2a-7e83-41f1-bd15-9a96dd6c7a1c.png)
![flatcar-2](https://user-images.githubusercontent.com/165034/156378616-44f468e6-4834-4209-9c98-9c6ce64ada65.png)
![flatcar-3](https://user-images.githubusercontent.com/165034/156378929-5cc30974-4a4f-4bb4-ba32-3fa128170cfa.png)

**Comment by jepio** [comment]
I haven't had the opportunity to test flatcar on xen yet, but remember seeing "error: not xen image" in my experiments with xen in the past when ubuntu switched to lz4 compression, which was not supported by the bootloader. So it might be something along those lines.

Could you try pvgrub2 instead?

**Comment by jepio** [comment]
I'm not sure Flatcar works in PV mode at all right now. Your best bet is to run in HVM mode.
